### PR TITLE
fix(Coachmark): accessibility improvements

### PR DESCRIPTION
**Pull request by jlongshore**
Contributes to [3489](https://github.com/carbon-design-system/ibm-products/issues/3489)

Accessibility review for the Coachmark suite of components. 

#### What did you change?
Minor styling and markup were changed

#### How did you test and verify your work?
Manually checked along with the IBM Accessibility browser plugin

### CODE REVIEW

**Comment by jlongshore** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by jlongshore** [comment]
recheck

**Comment by jlongshore** [comment]
recheck

**Comment by jlongshore** [comment]
recheck

**Comment by jlongshore** [comment]
recheck

File: packages/ibm-products/src/components/Coachmark/CoachmarkOverlay.js [hunk 1]
```diff
@@ -98,15 +107,17 @@ export let CoachmarkOverlay = forwardRef(
...
       >
         {isDraggable ? (
           <CoachmarkDragbar onDrag={handleDrag} onClose={onClose} />
         ) : (
           <CoachmarkHeader onClose={onClose} />
         )}
-        <div className={`${blockClass}__body`} ref={ref}>
-          {children}
+        <div className={`${blockClass}__body`} ref={ref} id={contentId}>
+          {getModifiedChildren()}
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/Coachmark/CoachmarkOverlay.js:120]: `getModifiedChildren()` and `modifyChildren()` are only referenced once each. Can they simply be inlined here?

File: packages/ibm-products/src/components/Coachmark/styles/_coachmark-overlay.scss [hunk 2]
```diff
@@ -202,7 +205,7 @@
 
     // BODY
     &__body {
-      padding: $spacing-01 $spacing-05 $spacing-05;
+      // padding: $spacing-01 $spacing-05 $spacing-05;
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/Coachmark/styles/_coachmark-overlay.scss:208]: Remove comment.

File: packages/ibm-products/src/components/CoachmarkBeacon/_coachmark-beacon.scss [hunk 3]
```diff
@@ -43,6 +46,26 @@
...
+            height: 18px;
+            border-radius: 100%;
+            content: '';
+            outline: $spacing-01 $focus solid;
+          }
+        }
+
+        // &:active {
+        //   outline: none;
+        // }
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/CoachmarkBeacon/_coachmark-beacon.scss:68]: Remove comment.

**Comment by elycheea** [comment]
@jlongshore is this PR ready for a re-review from @paul-balchin-ibm?

**Comment by jlongshore** [comment]
> @jlongshore is this PR ready for a re-review from @paul-balchin-ibm?

It should be good to review!

**Comment by paul-balchin-ibm** [comment]
Reviewing.

File: packages/ibm-products/src/components/Coachmark/styles/_coachmark-dragbar.scss [hunk 4]
```diff
@@ -16,53 +14,50 @@
...
-      }
-    }
-
-    &--close-btn {
-      width: $spacing-07;
-      height: $spacing-07;
-      margin-left: auto !important;
+    // &__light {
+    //   .#{$block-class}--close-btn {
+    //     &:active {
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/Coachmark/styles/_coachmark-dragbar.scss:24]: Delete comments.

File: packages/ibm-products/src/components/Coachmark/styles/_coachmark-header.scss [hunk 5]
```diff
@@ -20,36 +18,21 @@
...
-    &__dark {
-      #{$block-class}--close-btn {
-        &:active {
-          background-color: $button-tertiary-active;
-        }
-      }
-    }
+    // &__light {
+    //   #{$block-class}--close-btn {
+    //     &:active {
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/Coachmark/styles/_coachmark-header.scss:23]: Delete comments.

**Review by paul-balchin-ibm — CHANGES_REQUESTED** [review]
Just two "delete comments" requests and a question and we're good to go.

**Review by paul-balchin-ibm — APPROVED** [review]
LGTM 🎉

**Pull request merged**